### PR TITLE
RotateTool created which rotates most Layers correctly

### PR DESCRIPTION
**Pull request by oscaralmstrom**
(Ignore missleading branch name)

### CODE REVIEW

File: src/main/java/chalmers/pimp/model/IModel.java [hunk 1]
```diff
@@ -166,6 +166,36 @@
    */
   void moveActiveLayer(int dx, int dy);
 
+  /**
+   * Starts rotating the currently active layer. This method has no effect if there is no active
+   * layer.
+   *
+   */
+  void startRotatingActiveLayer(int x, int y);
```
Comment (albin-johansson) [inline src/main/java/chalmers/pimp/model/IModel.java]: No doc for params

File: src/main/java/chalmers/pimp/model/IRenderer.java [hunk 2]
```diff
@@ -78,11 +78,27 @@
...
   /**
-   * Sets the rotation.
+   * Creates and starts a new rotation transform. Will also save the previous transform.
+   *
+   * @param rotation   the rotation to use (degrees).
+   * @param startPoint the point for the coordinates.
+   * @param width      the width of the object.
+   * @param height     the height of the object.
+   */
+  void startTransform(double rotation, Point startPoint, int width, int height);
```
Comment (albin-johansson) [inline src/main/java/chalmers/pimp/model/IRenderer.java]: What happens when the point is null?

File: src/main/java/chalmers/pimp/model/ModelImpl.java [hunk 3]
```diff
@@ -191,6 +193,48 @@ public void moveActiveLayer(int dx, int dy) {
...
+    commandManager.insertCommand(cmd);
+    layerRotation = null;
+  }
+
+  @Override
+  public void rotateActiveLayer(double alpha) {
+    canvas.setActiveLayerRotation(alpha);
+    System.out.println(alpha);
+    notifyCanvasUpdateListeners();
+  }
```
Comment (albin-johansson) [inline src/main/java/chalmers/pimp/model/ModelImpl.java]: No printlns in production code.

File: src/main/java/chalmers/pimp/controller/IController.java [hunk 4]
```diff
@@ -14,17 +14,17 @@
...
 
   /**
-   * Sets a Transparent Pencil as selected tool in model
+   * Sets a transparent pencil as selected tool in model
    */
   void selectEraser();
 
   /**
-   * Sets Bucket as selected tool in model
+   * Sets s bucket as selected tool in model
```
Comment (theowiik) [inline src/main/java/chalmers/pimp/controller/IController.java]: minor typo..

File: src/main/java/chalmers/pimp/model/LayerRotation.java [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+ *
+ * @see IModel
+ */
+public final class LayerRotation {
+
+  private ModelMemento memento;
+  private boolean isFinished;
+  private Point rotationAnchorPoint;
+  private double mouseStartDegree;
+  private double baseDegree;
```
Comment (theowiik) [inline src/main/java/chalmers/pimp/model/LayerRotation.java]: what is a "baseDegree"?

**Pull request merged**